### PR TITLE
Various little code fixes.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: libr/util/spaces.c [hunk 1]
```diff
@@ -4,13 +4,7 @@
...
 R_API RSpaces *r_spaces_new(const char *name) {
 	RSpaces *sp = R_NEW0 (RSpaces);
-	if (!sp) {
-		return NULL;
-	}
-	if (!r_spaces_init (sp, name)) {
-		R_FREE (sp);
-		return NULL;
-	}
+	(void)r_spaces_init (sp, name);
```
Comment (radare) [inline libr/util/spaces.c]: wtf?

Comment (devnexen) [reply libr/util/spaces.c]: r_spaces_init free resources already if it fails at any stage.

Comment (radare) [reply libr/util/spaces.c]: if sp is null its crashing inside r_spaces_init because there's an assert in there

Comment (radare) [reply libr/util/spaces.c]: in case the resources are freed in r_spaces_init the sp is allocated but empty, aka not initialized, so it s wrong too

Comment (devnexen) [reply libr/util/spaces.c]: That s why I used the R_FREE macro

Comment (radare) [reply libr/util/spaces.c]: rfree sets the pointer to null, but you are passing this pointer as value not as reference, so it wont work

**Pull request merged**